hollow/src/main/java/com/netflix/hollow/core/read/OptionalBlobPartInput.java

```diff
@@ -18,27 +18,33 @@
 
 import com.netflix.hollow.core.memory.MemoryMode;
 import java.io.BufferedInputStream;
+import java.io.Closeable;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-public class OptionalBlobPartInput {
+public class OptionalBlobPartInput implements Closeable {
 
     private final Map<String, Object> inputsByPartName;
+    private final List<InputStream> streamsToClose;
 
     public OptionalBlobPartInput() {
         this.inputsByPartName = new HashMap<>();
+        this.streamsToClose = new ArrayList<>();
     }
     
     public void addInput(String partName, File file) {
         inputsByPartName.put(partName, file);
     }
 
     public void addInput(String partName, InputStream in) {
+        streamsToClose.add(in);
```

Comment:
Small preference for lettinig the layer that opens the streams also control whether they should be closed. So for eg. `getOptionalBlobPartInputs()` in `CinderBlobRetriever` could go from -
```
obpi.addInput(entry.getKey(), bis);
```
to 
```
obpi.addInput(entry.getKey(), bis);
obpi.streamsToClose.add(bis);
```
With the current implementation we'd only run into issues if we needed to close the `OptionalBlobPartInput` but reuse one of its input streams (and the stream has already been closed). I don't think we'll have that issue given how we use these streams. But this little tweak might be a bit cleaner and consistent with the rest of the code base.